advanced/Scripts/webpage.sh

```diff
@@ -393,13 +393,8 @@ ProcessDHCPSettings() {
         if [[ "${DHCP_LEASETIME}" == "0" ]]; then
             leasetime="infinite"
         elif [[ "${DHCP_LEASETIME}" == "" ]]; then
-            leasetime="24"
-            addOrEditKeyValPair "${setupVars}" "DHCP_LEASETIME" "${leasetime}"
-        elif [[ "${DHCP_LEASETIME}" == "24h" ]]; then
-            #Installation is affected by known bug, introduced in a previous version.
-            #This will automatically clean up setupVars.conf and remove the unnecessary "h"
-            leasetime="24"
-            addOrEditKeyValPair "${setupVars}" "DHCP_LEASETIME" "${leasetime}"
+            leasetime="24h"
+            addOrEditKeyValPair "${setupVars}" "DHCP_LEASETIME" "24"
         else
             leasetime="${DHCP_LEASETIME}h"
```

Comment:
> The only other case I can see that should be accounted for is when the var contains anything other than an integer. In that case I think we use the default `24h`.

To filter non-integer values, we could use here:


```suggestion
            leasetime="${DHCP_LEASETIME//[^0-9]/}h"
```

Reply:
Is that a POSIX solution?

Reply:
To be honest, I have no idea.

Reply:
I think this is just a variable expansion pattern, but I'm not sure if this is a POSIX solution.

Reply:
It's bash only: https://www.gnu.org/software/bash/manual/bash.html#Shell-Parameter-Expansion

Not listed in https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html

Reply:
OK.
We can use `sed`, but I'm sure there is a better choice than this:

```suggestion
            leasetime=$(echo "$DHCP_LEASETIME" | sed 's/[^0-9]*//g; s/$/h/')
```

Reply:
I think this PR covers what it says on the tin so we can look at changes in a separate PR.

Reply:
TBF, the rest of the function is not POSIX compatible - but that's for another time.

I think that there is probably little point in sanitising the value at this point. The web interface will only put integers into this setting - if someone has entered a bad value, then it should fall over 😉

Reply:
`leasetime=$(echo "$DHCP_LEASETIME" |  tr -cd [:digit:])`